src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,285 @@
+
+        for (Task task : tasks) {
+            System.out.print(tasks.indexOf(task) + 1 + ".");
+            task.printTask();
+        }
+        System.out.println("    ____________________________________________________________");
+    }
+
+
+
+    //completed tasks ( e.g. done 3)
+    public static void isCompleted(String command) {
+        int index = 0;
+        command = command.replace("done", " ");
+        command = command.strip(); //removes white space
+        try {
+            index = Integer.parseInt(command);
+        } catch (NumberFormatException e) {
+            System.out.println("Oh no! Please list a task number to be marked done :(");
+        }
+
+        try {
+            Task task = tasks.get(index - 1);
+            String oldTaskString = task.toString();
+            task.isDone = true;
+            updateFile(oldTaskString, task.toString());
+
+            System.out.println("    ____________________________________________________________");
+            System.out.println("    Nice! I've marked this task as done:");
+            System.out.print(index + ".");
+            task.printTask();
+            System.out.println("    ____________________________________________________________");
+        } catch (NullPointerException e) {
+            System.out.println("Oh no! Please enter a valid task number :(");
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+
+    //add task to list
+    public static void addTask(String line) {
+
+        if (line.startsWith("todo")) {
+            try {
+                line = line.substring(5);
+                addTodo(line,false);
+            } catch (StringIndexOutOfBoundsException e) {
+                System.out.println("OOPS!!! The description of a todo cannot be empty☹ Please try again!");
+                return;
+            }
+        } else if (line.startsWith("deadline")) {
+            try {
+                line = line.substring(9);
+                addDeadline(line,false);
+            } catch (StringIndexOutOfBoundsException e) {
+                System.out.println("OOPS!!! The description of a deadline cannot be empty☹ Please try again! ");
+                return;
+            }
+        } else if (line.startsWith("event")) {
+            try {
+                line = line.substring(6);
+                addEvent(line,false);
+            } catch (StringIndexOutOfBoundsException e) {
+                System.out.println("OOPS!!! The description of a event cannot be empty☹ Please try again!");
+                return;
+            }
+        } else {
+            new Exception("not valid");
+            return;
+        }
+
+        System.out.println("Got it. I've added this task:");
+        tasks.get(tasks.size()-1).printTask();
+        System.out.println("Now you have " + tasks.size() + " tasks in the list!");
+    }
+
+    public static void toDeleteTask(String line){
```

Comment:
The function `toDeleteTask()` can be changed to `deleteTask()`. Typically, function names do not start with `toDoSomething()` but instead `doSomething()`.